plugin/src/test/java/io/jenkins/plugins/casc/jmh/benchmarks/SecretSourceResolverBenchmark.java

```diff
@@ -0,0 +1,47 @@
+package io.jenkins.plugins.casc.jmh.benchmarks;
+
+import io.jenkins.plugins.casc.ConfigurationContext;
+import io.jenkins.plugins.casc.ConfiguratorRegistry;
+import jenkins.benchmark.jmh.JmhBenchmark;
+import jenkins.benchmark.jmh.JmhBenchmarkState;
+import org.junit.contrib.java.lang.system.EnvironmentVariables;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Mode;
+
+@JmhBenchmark
+@BenchmarkMode(Mode.Throughput)
+public class SecretSourceResolverBenchmark {
+    public static class JenkinsState extends JmhBenchmarkState {
+        private final EnvironmentVariables environmentVariables = new EnvironmentVariables();
+        private ConfigurationContext context = null;
+
+        @Override
+        public void setup() throws Exception {
+            ConfiguratorRegistry registry = ConfiguratorRegistry.get();
+            context = new ConfigurationContext(registry);
+            environmentVariables.set("FOO", "HELLO");
+            environmentVariables.set("BAR", "WROLD");
+        }
+    }
+
+    @Benchmark
+    public void emptyString(JenkinsState state) {
+        state.context.getSecretSourceResolver().resolve("");
```

Comment:
@jetersen I think you should pass this to a `Blackhole` to avoid dead-code elimination: http://tutorials.jenkov.com/java-performance/jmh.html#dead-code-elimination

Reply:
Your absolutely right, I found a clear optimization yet my benchmark was made worse :|

Reply:
Numbers a still good, now even with the blackhole

![image](https://user-images.githubusercontent.com/1661688/83355628-ed33bc00-a360-11ea-9256-255c2c258ce7.png)

Reply:
FYI this speed increase is definitely noticeable 👏